execution/binary/table.go

```diff
@@ -201,6 +199,42 @@ func newOperation(expr parser.ItemType, vectorBinOp bool) (operation, error) {
 	return nil, parse.UnsupportedOperationErr(expr)
 }
 
+// histogramFloatOperation is an operation defined one histogram and one float.
+type histogramFloatOperation func(lhsHist *histogram.FloatHistogram, rhsFloat float64) *histogram.FloatHistogram
+
+func undefinedHistogramOp(_ *histogram.FloatHistogram, _ float64) *histogram.FloatHistogram {
+	return nil
+}
+
+var lhsHistogramOperations = map[string]histogramFloatOperation{
+	"*": func(hist *histogram.FloatHistogram, float float64) *histogram.FloatHistogram {
+		return hist.Copy().Mul(float)
+	},
+	"/": func(hist *histogram.FloatHistogram, float float64) *histogram.FloatHistogram {
+		return hist.Copy().Div(float)
+	},
+}
+
+var rhsHistogramOperations = map[string]histogramFloatOperation{
+	"*": func(hist *histogram.FloatHistogram, float float64) *histogram.FloatHistogram {
+		return hist.Copy().Mul(float)
+	},
+}
+
+func getHistogramFloatOperation(expr parser.ItemType, scalarSide ScalarSide) histogramFloatOperation {
+	t := parser.ItemTypeStr[expr]
+	var operation histogramFloatOperation
+	if scalarSide == ScalarSideLeft {
+		operation = rhsHistogramOperations[t]
+	} else {
+		operation = lhsHistogramOperations[t]
+	}
+	if operation != nil {
+		return operation
+	}
+	return undefinedHistogramOp
```

Comment:
shouldnt that return something like "unimplemented" to trigger promql fallback?

Reply:
The fallback can only be applied at plan compilation time when we detect that a function or operator is implemented in the engine. In this specific case, we want to cover operators that are simply not "defined" between a histogram and float operand, even in Prometheus. 

For example, something like a % b is only defined when both a and b are floats. But to know their type, we have to actually execute the query and decode chunks. To further complicate matters, a series can have interleaved histograms and floats, so for some samples the operation will be defined and for others it won't. 

So bottom line, it should be fine to produce a nil value when an operation between a histogram and float is simply not defined, as in the a % b example.

Reply:
ah! ok i understand; can you add a small test to ensure that we are behaving like prometheus in that case?

Reply:
nvm, i have found 
```
If either operator has to sum up a mix of histogram samples and float samples, the corresponding vector element is removed from the output vector entirely.
```

Reply:
ok, i think this will just work!